src/main/kotlin/gg/essential/elementa/components/ScrollComponent.kt

```diff
@@ -800,6 +806,30 @@ class ScrollComponent constructor(
 
     }
 
+    /**
+     * Constraints the scrollbar grip's height to be a certain minimum height, or the [desiredHeight].
+     * This is the default constraint for vertical scrollbar grips if [limitScrollBarGripHeight] is enabled.
+     *
+     * @param desiredHeight The intended height for the scrollbar grip.
+     */
+    inner class ScrollBarGripMinHeightConstraint(
+        private val desiredHeight: HeightConstraint
+    ) : HeightConstraint {
+        override var cachedValue: Float = 0f
+        override var recalculate: Boolean = true
+        override var constrainTo: UIComponent? = null
+
+        override fun getHeightImpl(component: UIComponent): Float {
+            val parent = component.parent
+            val minimumHeight = if (parent.getHeight() < 200) { 15.percent } else { 10.percent }
+            return desiredHeight.coerceAtLeast(minimumHeight).getHeight(component)
```

Comment:
This creates a bunch of constraints every call only to evaluate them once before dropping them again. Why not just work with the Float values directly?